spf.go

```diff
@@ -776,19 +776,27 @@ func (r *resolution) mxField(res Result, field, domain string) (bool, Result, er
 	}
 
 	mxips := []net.IP{}
+	var tempErr error
+
 	for _, mx := range mxs {
 		ips, err := r.resolver.LookupIPAddr(r.ctx, mx.Host)
 		if err != nil {
 			// https://tools.ietf.org/html/rfc7208#section-5
 			if isTemporary(err) {
-				return true, TempError, err
+				tempErr = err
+				continue
```

Comment:
I'd suggest a slightly different approach, since a DNS temporary error may be something else than RCODE 3.

What about leaving this as-is, but do the following right before?

```go
// If the address of the MX record was not found, we just skip it.
// https://tools.ietf.org/html/rfc7208#section-5
if dnsErr, ok := err.(*net.DNSError); ok && dnsErr.IsNotFound {
    continue
}
if isTemporary(err) {
    return true, TempError, err
}
```

Then, no extra check is needed below, and I think this behaviour is more in line with the RFC.

What do you think?

Reply:
I have changed the commit to your suggestion and fixed the testcases to match only the isnotfound error